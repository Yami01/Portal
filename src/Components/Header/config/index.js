/**
 *
 * @format
 * @flow
 *
 */

export const TITLE_ICON_HEIGHT = 24

export const LEFT_BUTTON_CONTAINER_TEST_ID = 'leftButtonContainerTestID'
export const RIGHT_BUTTON_CONTAINER_TEST_ID = 'rightButtonContainerTestID'
export const TITLE_ICON_TEXT_TEST_ID = 'titleIconTextTestID'
