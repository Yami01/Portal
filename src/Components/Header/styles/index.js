/**
 *
 * @format
 * @flow
 *
 */

import { StyleSheet } from 'react-native'

import type { HeaderStyleType } from '../types'

const HeaderStyles: HeaderStyleType = StyleSheet.create({
  buttonContainer: {
    minWidth: 24,
    height: 24,
  },
  buttonContainerLeft: {
    marginRight: 12,
  },
  buttonContainerLeftOnRight: {
    marginLeft: 12,
    marginRight: 4,
  },
  buttonContainerRight: {
    marginLeft: 12,
  },
  container: {
    flexShrink: 1,
    flexGrow: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  titleContainer: {
    flexShrink: 1,
    flexGrow: 1,
    flexDirection: 'row',
  },
  progressBarContainer: {
    width: '60%',
    right: 0,
    top: 8,
    position: 'absolute',
    alignSelf: 'center',
  },
})

export default HeaderStyles
