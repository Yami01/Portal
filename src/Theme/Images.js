/**
 *
 * @return {*}
 */
export default function () {
  return {
    logo: require('@/Assets/Images/TOM.png'),
  }
}
