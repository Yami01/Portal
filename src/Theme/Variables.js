/**
 * This file contains the application's variables.
 *
 * Define color, sizes, etc. here instead of duplicating them throughout the components.
 * That allows to change them more easily later on.
 */

/**
 * Colors
 */
export const Colors = {
  // Example colors:
  transparent: 'rgba(0,0,0,0)',
  inputBackground: '#FFFFFF',
  white: '#ffffff',
  text: '#212529',
  primary: '#E14032',
  success: '#28a745',
  error: '#dc3545',
  black: '#000000',
  dark_gray: '#363e3f',
  gray: '#596c80',
  gray_650: '#5b636b',
  gray_250: '#d6d8da',
  gray_750: '#333d47',
  gray_950: '#5c646c',
  red_950: '#e10c0a',
}

export const NavigationColors = {
  primary: Colors.primary,
}

/**
 * FontSize
 */
export const FontSize = {
  small: 16,
  regular: 20,
  large: 40,
}

/**
 * Metrics Sizes
 */
const tiny = 4 // 8
const small = tiny * 2 // 8
const regular = tiny * 3 // 12
const large = regular * 2 // 24
export const MetricsSizes = {
  tiny,
  small,
  regular,
  large,
}

export default {
  Colors,
  NavigationColors,
  FontSize,
  MetricsSizes,
}
