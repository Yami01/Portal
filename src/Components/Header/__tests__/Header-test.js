/**
 *
 * Header Test
 * @format
 * @flow
 *
 */

import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'
import Header from '../Header'
import {
  LEFT_BUTTON_CONTAINER_TEST_ID,
  RIGHT_BUTTON_CONTAINER_TEST_ID,
  TITLE_ICON_TEXT_TEST_ID,
} from '../config'
import { Colors } from '@/Theme/Variables'

describe('Header', () => {
  const onPressMock = jest.fn()

  const leftButtonAccessibilityLabel = 'Accessible header button left'
  const leftButton = {
    iconFillColor: Colors.white,
    iconGroup: 'general',
    iconName: 'menu',
  }

  const rightButtonAccessibilityLabel = 'Accessible header button right'
  const rightButtonLabel = 'Updated today'
  const rightButton = {
    iconFillColor: Colors.white,
    iconGroup: 'general',
    iconName: 'notifications',
    badgeNumber: 1,
  }

  const titleLabel = 'AIA Vitality'
  const titleIconProps = {
    titleIconGroup: 'general',
    titleIconName: 'logo_white',
  }

  const testID = 'Header Test ID'

  test('render with title label and side buttons', () => {
    const tree = render(
      <Header
        leftButton={leftButton}
        titleLabel={titleLabel}
        rightButton={rightButton}
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('render with title label and side buttons on right', () => {
    const tree = render(
      <Header
        leftButton={leftButton}
        titleLabel={titleLabel}
        rightButton={rightButton}
        isAllButtonsOnRight
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Render logo only', () => {
    const element = render(<Header {...titleIconProps} />).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('Render title with icon', () => {
    const element = render(<Header {...titleIconProps} />).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('Render left button and logo only', () => {
    const element = render(
      <Header leftButton={leftButton} {...titleIconProps} />,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('Render right button with button label and title logo only', () => {
    const element = render(
      <Header
        {...titleIconProps}
        rightButton={{
          ...rightButton,
          label: rightButtonLabel,
        }}
      />,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('Render with additional configs', () => {
    const element = render(
      <Header
        titleLabel={titleLabel}
        extraPaddingBottom={30}
        isLargePadding
        shouldIgnoreStatusBarHeight
        withShadow
      />,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })

  test('Not render button containers if no buttons', () => {
    const { queryByTestId } = render(<Header titleLabel={titleLabel} />)

    expect(queryByTestId(LEFT_BUTTON_CONTAINER_TEST_ID)).toBeNull()
    expect(queryByTestId(RIGHT_BUTTON_CONTAINER_TEST_ID)).toBeNull()
  })

  test('Not render right button container if no right button and align to left', () => {
    const { queryByTestId } = render(
      <Header
        titleLabel={titleLabel}
        leftButton={leftButton}
        isTitleAlignLeft
      />,
    )

    expect(queryByTestId(LEFT_BUTTON_CONTAINER_TEST_ID)).not.toBeNull()
    expect(queryByTestId(RIGHT_BUTTON_CONTAINER_TEST_ID)).toBeNull()
  })

  test('render right button container even no right button but align to center', () => {
    const { queryByTestId } = render(
      <Header titleLabel={titleLabel} leftButton={leftButton} />,
    )

    expect(queryByTestId(LEFT_BUTTON_CONTAINER_TEST_ID)).not.toBeNull()
    expect(queryByTestId(RIGHT_BUTTON_CONTAINER_TEST_ID)).not.toBeNull()
  })

  test('render left button only if isOnlyLeftIcon true', () => {
    const { queryByTestId } = render(
      <Header
        titleLabel={titleLabel}
        leftButton={leftButton}
        rightButton={rightButton}
        testID={testID}
        isAllButtonsOnRight
        isTitleAlignLeft
        isOnlyLeftIcon
      />,
    )

    expect(queryByTestId(LEFT_BUTTON_CONTAINER_TEST_ID)).not.toBeNull()
    expect(queryByTestId(RIGHT_BUTTON_CONTAINER_TEST_ID)).toBeNull()
    expect(queryByTestId(TITLE_ICON_TEXT_TEST_ID)).toBeNull()
    const container = queryByTestId(testID)
    expect(container).toHaveStyle({ backgroundColor: Colors.transparent })
  })

  test('Test if header left button is clickable', () => {
    const { getByA11yLabel } = render(
      <Header
        leftButton={{
          ...leftButton,
          onPress: onPressMock,
        }}
      />,
    )

    const button = getByA11yLabel(leftButtonAccessibilityLabel)
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalled()
  })

  test('Test if header right button is clickable', () => {
    const { getByA11yLabel } = render(
      <Header
        rightButton={{
          ...rightButton,
          onPress: onPressMock,
        }}
      />,
    )

    const button = getByA11yLabel(rightButtonAccessibilityLabel)
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalled()
  })

  test('Test if title is clickable', () => {
    const { getByTestId } = render(
      <Header titleLabel={titleLabel} titleOnPress={onPressMock} />,
    )

    const button = getByTestId(TITLE_ICON_TEXT_TEST_ID)
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalled()
  })
})
