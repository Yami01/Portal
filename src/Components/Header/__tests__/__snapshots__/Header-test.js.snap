// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render left button and logo only 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginRight": 12,
          },
        ]
      }
      testID="leftButtonContainerTestID"
    >
      <View>
        <View
          accessibilityHint="This button will take you home"
          accessibilityLabel="Accessible header button left"
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
            testID="touchableWrapperTestID"
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <svg
                data-file-name="SvgMenu"
                fill="#fff"
                height={24}
                width={24}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          />
          <View
            accessibilityHint="This is header title icon"
            accessibilityLabel="Accessible header title icon"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <svg
              data-file-name="SvgLogoWhite"
              fill="#fff"
              height={24}
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginLeft": 12,
          },
        ]
      }
      testID="rightButtonContainerTestID"
    />
  </View>
</View>
`;

exports[`Header Render logo only 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          />
          <View
            accessibilityHint="This is header title icon"
            accessibilityLabel="Accessible header title icon"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <svg
              data-file-name="SvgLogoWhite"
              fill="#fff"
              height={24}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header Render right button with button label and title logo only 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginRight": 12,
          },
        ]
      }
      testID="leftButtonContainerTestID"
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          />
          <View
            accessibilityHint="This is header title icon"
            accessibilityLabel="Accessible header title icon"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <svg
              data-file-name="SvgLogoWhite"
              fill="#fff"
              height={24}
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginLeft": 12,
          },
        ]
      }
      testID="rightButtonContainerTestID"
    >
      <View>
        <View
          accessibilityHint="This button will take to the notification"
          accessibilityLabel="Accessible header button right"
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                false,
              ]
            }
            testID="touchableWrapperTestID"
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 12,
                    "marginRight": 5,
                  },
                ]
              }
            >
              Updated today
            </Text>
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <svg
                data-file-name="SvgNotifications"
                fill="#fff"
                height={24}
                width={24}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 7.5,
                "height": 15,
                "width": 15,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#d31145",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "position": "absolute",
                  "right": -3,
                  "top": -3,
                  "zIndex": 10,
                },
                Object {
                  "backgroundColor": "#e10c0a",
                  "borderColor": "#fff",
                  "borderWidth": 1,
                },
              ],
            ]
          }
          testID="badgeTestID"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                  },
                  undefined,
                ],
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header Render title with icon 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          />
          <View
            accessibilityHint="This is header title icon"
            accessibilityLabel="Accessible header title icon"
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <svg
              data-file-name="SvgLogoWhite"
              fill="#fff"
              height={24}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header Render with additional configs 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 50,
        },
        "paddingTop": 20,
      },
      Object {
        "elevation": 12,
        "shadowColor": "#363e3f",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          >
            AIA Vitality
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header render with title label and side buttons 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginRight": 12,
          },
        ]
      }
      testID="leftButtonContainerTestID"
    >
      <View>
        <View
          accessibilityHint="This button will take you home"
          accessibilityLabel="Accessible header button left"
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
            testID="touchableWrapperTestID"
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <svg
                data-file-name="SvgMenu"
                fill="#fff"
                height={24}
                width={24}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          >
            AIA Vitality
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginLeft": 12,
          },
        ]
      }
      testID="rightButtonContainerTestID"
    >
      <View>
        <View
          accessibilityHint="This button will take to the notification"
          accessibilityLabel="Accessible header button right"
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                false,
              ]
            }
            testID="touchableWrapperTestID"
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <svg
                data-file-name="SvgNotifications"
                fill="#fff"
                height={24}
                width={24}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 7.5,
                "height": 15,
                "width": 15,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#d31145",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "position": "absolute",
                  "right": -3,
                  "top": -3,
                  "zIndex": 10,
                },
                Object {
                  "backgroundColor": "#e10c0a",
                  "borderColor": "#fff",
                  "borderWidth": 1,
                },
              ],
            ]
          }
          testID="badgeTestID"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                  },
                  undefined,
                ],
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Header render with title label and side buttons on right 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#d31145",
        "borderBottomLeftRadius": 16,
        "flexDirection": "row",
        "overflow": "visible",
        "paddingHorizontal": 16,
        "width": "100%",
        "zIndex": 98,
      },
      Object {
        "backgroundColor": "#d31145",
        "paddingBottom": AnimatedValue {
          " __value": 16,
        },
        "paddingTop": 36,
      },
      false,
      undefined,
    ]
  }
>
  <View
    accessibilityHint="This is the header"
    accessibilityLabel="Accessible Header"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
          },
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        accessibilityRole="text"
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
        testID="titleIconTextTestID"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                undefined,
                Array [
                  Object {
                    "fontSize": 17,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              ]
            }
          >
            AIA Vitality
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginLeft": 12,
            "marginRight": 4,
          },
        ]
      }
      testID="leftButtonContainerTestID"
    >
      <View>
        <View
          accessibilityHint="This button will take you home"
          accessibilityLabel="Accessible header button left"
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
            testID="touchableWrapperTestID"
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <svg
                data-file-name="SvgMenu"
                fill="#fff"
                height={24}
                width={24}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 24,
            "minWidth": 24,
          },
          Object {
            "marginLeft": 12,
          },
        ]
      }
      testID="rightButtonContainerTestID"
    >
      <View>
        <View
          accessibilityHint="This button will take to the notification"
          accessibilityLabel="Accessible header button right"
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                false,
              ]
            }
            testID="touchableWrapperTestID"
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                ]
              }
            >
              <svg
                data-file-name="SvgNotifications"
                fill="#fff"
                height={24}
                width={24}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderRadius": 7.5,
                "height": 15,
                "width": 15,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#d31145",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "position": "absolute",
                  "right": -3,
                  "top": -3,
                  "zIndex": 10,
                },
                Object {
                  "backgroundColor": "#e10c0a",
                  "borderColor": "#fff",
                  "borderWidth": 1,
                },
              ],
            ]
          }
          testID="badgeTestID"
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                  },
                  undefined,
                ],
              ]
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
